src/docs/sdk/performance/dynamic-sampling-context.mdx

```diff
@@ -0,0 +1,124 @@
+
+SDKs must not add other vendors' baggage from incoming requests to outgoing requests.
+Sentry SDKs only concern themselves with Sentry baggage.
+
+</Alert>
+
+The following is an example of what a baggage header containing Dynamic Sampling Context may look like:
+
+```
+baggage: other-vendor-value-1=foo;bar;baz, sentry-traceid=771a43a4192642f0b136d5159a501700, sentry-publickey=49d0f7386ad645858ae85020e393bef3; sentry-userid=Am%C3%A9lie, other-vendor-value-2=foo;bar;
+```
+
+See the [Payloads section](#payloads) for a complete list of key-value pairs that SDKs should propagate.
+
+## Payloads
+
+Dynamic Sampling Context is propagated via a baggage header and sent to Sentry via <Link to="/sdk/envelopes/#transaction">transaction envelope headers</Link>.
+
+### Baggage-Header
+
+SDKs may set the following key-value pairs on baggage headers.
+While all of these values are optional, SDKs should make their best effort to add as many of them to the baggage header as possible when starting a trace.
+
+- `sentry-traceid` - The original trace ID as generated by the SDK
+- `sentry-publickey` - Public key as defined by the user via the DSN in the SDK options
+- `sentry-release` - The release as defined by the user in the SDK options
+- `sentry-environment` - The environment as defined by the user in the SDK options
+- `sentry-transaction` - The name of the trace's origin transaction in unparameterized (raw) format
+- `sentry-userid` - User ID as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>
+- `sentry-usersegment` - User segment as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>
+- `sentry-samplerate` - Sample rate as defined by the user in the SDK options
+
+SDKs must set all of the keys in the form of "`sentry-[name]`".
+The delimiter "`sentry-`" acts to identify key-value pairs set by Sentry SDKs.
+Additionally, we chose `[name]` to be written in "snake case" without any underscore ( `_` ) characters. This naming convention is the most language agnostic.
+
+### Envelope Header
+
+Dynamic Sampling Context is transferred to Sentry through the <Link to="/sdk/envelopes/#transaction">transaction envelope headers</Link>, keyed by `trace`.
+It corresponds directly to the definition of <Link to="/sdk/performance/trace-context/#trace-context">Trace Context</Link>.
+
+When a transaction is reported to Sentry, the Dynamic Sampling Context must be mapped to Trace Context in the following way:
+
+- `sentry-release` ➝ `release`
+- `sentry-environment` ➝ `environment`
+- `sentry-transaction` ➝ `transaction`
+- `sentry-userid` ➝ `user.id`
+- `sentry-usersegment` ➝ `user.segment`
+- `sentry-samplerate` ➝ `sample_rate`
+- `sentry-traceid` ➝ `trace_id`
+- `sentry-publickey` ➝ `public_key`
+
+## Unified Propagation Mechanism
+
+Todo:
+
+- incoming request algorithm
+- outgoing request algorithm
+
+## Considerations
+
+Todo:
+
+- Why baggage and not trace context https://www.w3.org/TR/trace-context/?
```

Comment:
Trace context is more restrictive in terms of size than baggage. Also baggage is more flexible in terms of encoding and characters according to our internal document on the decision.